src/VisualStudio/Core/Def/ProjectSystem/IEntryPointFinderService.cs

```diff
@@ -19,4 +19,11 @@ internal interface IEntryPointFinderService : ILanguageService
     /// <param name="symbol">The namespace to search.</param>
     /// <param name="findFormsOnly">Restrict the search to only Windows Forms classes. Note that this is only implemented for VisualBasic</param>
     IEnumerable<INamedTypeSymbol> FindEntryPoints(INamespaceSymbol symbol, bool findFormsOnly);
+
+    /// <summary>
+    /// Finds the types that contain entry points like the Main method in a given compilation.
+    /// </summary>
+    /// <param name="compilation">The compilation to search.</param>
+    /// <param name="findFormsOnly">Restrict the search to only Windows Forms classes. Note that this is only implemented for VisualBasic</param>
+    IEnumerable<INamedTypeSymbol> FindEntryPoints(Compilation compilation, bool findFormsOnly);
```

Comment:
FYI @drewnoakes here's an improved method for you to call here:

https://github.com/dotnet/project-system/blob/05aa25172c582d92041fdac9b15ee826e28ec1ef/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/StartupObjectsEnumProvider.cs#L87
https://github.com/dotnet/project-system/blob/05aa25172c582d92041fdac9b15ee826e28ec1ef/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/VisualBasic/SplashScreenEnumProvider.cs#L86

The .GlobalNamespace line isn't right, so we've been working around it all these years. A cleanup on our side to deal with this then broke stuff.

Reply:
Thanks! Filed https://github.com/dotnet/project-system/issues/9676